src/locale/localed.c

```diff
@@ -395,6 +398,30 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
         for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++)
                 free_and_replace(c->locale[p], new_locale[p]);
 
+        /* Generate locale in case it is missing and the system is using locale-gen */
+        if (use_localegen && (locale_is_installed(c->locale[VARIABLE_LANG]) <= 0)) {
```

Comment:
hmm, what if a user passes complex locale settings to us? i.e. with all fields set differently? shouldn't we generate locales for all of them? i.e. shouldn't this be a loop of LocaleVariable type from 0 to _VARIABLE_LC_MAX

Reply:
Yes - with locale_is_installed() being fast enough, that will work. Hopefully nobody actually does set every value to a different locale, because that would cause locale-gen to really run for a while, taking more than 2min of time.